core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -131,11 +120,46 @@ public SetupWizard(Jenkins j) throws IOException, InterruptedException {
             throw new AssertionError(e);
         }
     }
+    
+    /**
+     * Called during the initial setup to create an admin user
+     */
+    public void doCreateAdminUser(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        Jenkins j = Jenkins.getInstance();
+        j.checkPermission(Jenkins.ADMINISTER);
+        
+        // This will be set up by default. if not, something changed, ok to fail
+        HudsonPrivateSecurityRealm securityRealm = (HudsonPrivateSecurityRealm)j.getSecurityRealm();
+        
+        User admin = securityRealm.getUser(SetupWizard.initialSetupAdminUserName);
+        try {
+            if(admin != null) {
+                admin.delete(); // assume the new user may well be 'admin'
+            }
+            
+            User u = securityRealm.createAccountByAdmin(req, rsp, "/jenkins/install/SetupWizard/setupWizardFirstUser.jelly", j.getRootUrl());
```

Comment:
Use `sendRedirect2` and pass `/`. Do not use `getRootUrl` internally. [Guide](https://wiki.jenkins-ci.org/display/JENKINS/Hyperlinks+in+HTML)